logger/plugin.cc

```diff
@@ -0,0 +1,141 @@
+/**
+*		Tempesta FW
+*
+* Copyright (C) 2025 Tempesta Technologies, Inc.
+*
+* This program is free software; you can redistribute it and/or modify it
+* under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 2 of the License,
+* or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+* FITNESS FOR A PARTICULAR PURPOSE.
+* See the GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License along with
+* this program; if not, write to the Free Software Foundation, Inc., 59
+* Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+*/
+
+#include <dlfcn.h>
+#include <spdlog/spdlog.h>
+#include <fmt/format.h>
+
+#include "../libtus/error.hh"
+#include "event_processor.hh"
+#include "plugin.hh"
+
+Plugin::Plugin(const std::string &plugin_path,
+	       const ClickHouseConfig& config,
+	       std::atomic<bool> *stop_flag)
+{
+	handle_ = dlopen(plugin_path.c_str(),
+			 RTLD_LAZY | RTLD_LOCAL | RTLD_DEEPBIND);
+	if (!handle_)
+		throw tus::Except("Failed to load plugin {}: {}",
+				  plugin_path, dlerror());
+
+	auto get_api = reinterpret_cast<TfwLoggerPluginGetApiFunc>(
+		dlsym(handle_, "get_plugin_api"));
+
+	if (!get_api)
+	{
+		dlclose(handle_);
+		handle_ = nullptr;
+		throw tus::Except("Plugin {} missing get_plugin_api function",
+				  plugin_path);
+	}
+
+	api_ = get_api();
+	if (!api_ || api_->version != TFW_PLUGIN_VERSION)
+	{
```

Comment:
```suggestion
	if (!api_ || api_->version != TFW_PLUGIN_VERSION) {
```

Please fix all other similar places